src/workflow_tasks.rs

```diff
@@ -266,14 +271,28 @@ impl WorkflowTaskManager {
             legacy_query
         };
 
+        // Evict all run ids that were invalidated by the cache eviction policy and have no pending/outstanding activations or workflow tasks.
+        for run_id in self.workflow_cache_manager.read().get_pending_evictions() {
+            if !self.pending_activations.has_pending(&run_id)
+                && !self.outstanding_activations.contains_key(&run_id)
+                && !self.outstanding_workflow_tasks.contains_key(&run_id)
+            {
+                self.evict_run(&run_id);
+            }
+        }
```

Comment:
Need some UTs for this part. One problem I think I see here is if the run for which we are currently applying a poll response is in the pending evictions list we'll issue an eviction and then immediately issue the new activation, which works but is unnecessary. `touch` should probably be at the top of this method.

Also worth extracting this bit to it's own little function for readability

Reply:
This is removed as I've inverted where touch/evict is done.